Packages/com.unity.inputsystem/CHANGELOG.md

```diff
@@ -69,11 +69,32 @@ however, it has to be formatted properly to pass verification tests.
 ### Added
 
 - Added support for "Hori Co HORIPAD for Nintendo Switch", "HORI Pokken Tournament DX Pro Pad", "HORI Wireless Switch Pad", "HORI Real Arcade Pro V Hayabusa in Switch Mode", "PowerA NSW Fusion Wired FightPad", "PowerA NSW Fusion Pro Controller (USB only)", "PDP Wired Fight Pad Pro: Mario", "PDP Faceoff Wired Pro Controller for Nintendo Switch", "PDP Faceoff Wired Pro Controller for Nintendo Switch", "PDP Afterglow Wireless Switch Controller", "PDP Rockcandy Wired Controller".
+- Added support for SteelSeries Nimbus+ gamepad on Mac (Addition contributed by [Mollyjameson](https://github.com/MollyJameson)).
 - Added a new `DeltaControl` control type that is now used for delta-style controls such as `Mouse.delta` and `Mouse.scroll`.
   * Like `StickControl`, this control has individual `up`, `down`, `left`, and `right` controls (as well as `x` and `y` that it inherits from `Vector2Control`). This means it is now possible to directly bind to individual scroll directions (such as `<Mouse>/scroll/up`).
 
-### Added
-- Added support for SteelSeries Nimbus+ gamepad on Mac (Addition contributed by [Mollyjameson](https://github.com/MollyJameson)).
+#### Actions
+
+- Added new APIs for getting and setting parameter values on interactions, processors, and composites.
+  ```CSharp
+  // Get parameter.
+  action.GetParameterValue("duration");     // Any "duration" value on any binding.
+  action.GetParameterValue("tap:duration"); // "duration" on "tap" interaction on any binding.
```

Comment:
> Would e.g. action.GetParameterValue<TapInteraction>("duration")

I considered this and discarded it in favor of the current approach. Interactions, processors, and composites are referred to in the API by their registration names, not by their implementation type -- which really is an implementation detail.

> if string based there needs to be somewhere to lookup the valid values here and it would be good to provide docs for how to approach figuring out what to write in that string and what naming scheme/syntax rules apply.

The docs mention this is the same as the registration name.

```
// "tap" referred to by name.
new InputAction(binding: "<Gamepad>/buttonSouth", interactions: "tap(duration=0.4)"
```

> should I expect "tap:duration" to affect taps, multi-tap and slow-tap or only what is corresponding to TapInteraction?

Only a single interaction (`TapInteraction`) has "tap" as its registration name.

>  if I set this and add processors/interactions after the e.g. parameter override has been applied, is it also applied to the newly added interactions/processors?

Yes, stated in the docs (API and manual).

Reply:
OK, guess I have overlooked where we describe currently available interaction string names in the manual/docs then, e.g. a list containing "tap" (among others) as a reference list of other valid options, where is this documented currently?

Reply:
It's documented in `Interactions.md`, though probably not nearly as clearly as it could be.

As per @andrew-oc's suggestion I added overloads that allow using C# expressions to specify this information instead:

```
action.GetParameterValue((TapInteraction x) => x.duration);

action.ApplyParameterOverride((TapInteraction x) => x.duration, 0.4f);
```

That has a number of advantages. Gets rid of the registration name. Provides a value type and thus type-safety. And gets rid of the string parameter name.

Reply:
Thats nice